modules/ROOT/pages/values-and-types/lists.adoc

```diff
@@ -34,274 +32,350 @@ RETURN [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] AS list
+  (matrix4:Movie {title: 'The Matrix Resurrections', released: 2021}),
+  (keanu)-[:ACTED_IN]->(johnnymnemonic),
+  (keanu)-[:ACTED_IN]->(thematrixrevolutions),
+  (keanu)-[:ACTED_IN]->(thematrixreloaded),
+  (keanu)-[:ACTED_IN]->(thereplacements),
+  (keanu)-[:ACTED_IN]->(thematrix),
+  (keanu)-[:ACTED_IN]->(thedevilsadvocate),
+  (keanu)-[:ACTED_IN]->(matrix4)
+----
+
+=== Examples
+
+This example returns a list that contains the year when the movies was released.
+The pattern matching in the pattern comprehension looks for `Matrix` in the movie title and that the node `a` (`Person` node with the name `Keanu Reeves`) has a relationship with the movie.
 
 .Query
 [source, cypher]
 ----
-RETURN [x IN range(0,10) WHERE x % 2 = 0 | x^3 ] AS result
+MATCH (keanu:Person {name: 'Keanu Reeves'})
+RETURN [(keanu)-->(b:Movie) WHERE b.title CONTAINS 'Matrix' | b.released] AS years
 ----
 
 .Result
 [role="queryresult",options="header,footer",cols="1*<m"]
 |===
-| +result+
-| +[0.0,8.0,64.0,216.0,512.0,1000.0]+
+| +years+
+| +[2021,2003,2003,1999]+
 1+d|Rows: 1
 |===
 
-Either the `WHERE` part, or the expression, can be omitted, if you only want to filter or map respectively.
+The whole predicate, including the `WHERE` keyword, is optional and may be omitted.
+
+This example returns a sorted list that contains years.
+The pattern matching in the pattern comprehension looks for movie nodes that has a relationship with the node `a` (`Person` node with the name `Keanu Reeves`).
 
 .Query
 [source, cypher]
 ----
-RETURN [x IN range(0,10) WHERE x % 2 = 0 ] AS result
+MATCH (keanu:Person {name: 'Keanu Reeves'})
+WITH [(keanu)-->(b:Movie) | b.released] AS years
+UNWIND years AS year
+WITH year ORDER BY year
+RETURN COLLECT(year) AS sorted_years
 ----
 
 .Result
 [role="queryresult",options="header,footer",cols="1*<m"]
 |===
-| +result+
-| +[0,2,4,6,8,10]+
+| +sorted_years+
+| +[1995,1997,1999,2000,2003,2003,2021]+
 1+d|Rows: 1
 |===
 
+=== Storing lists as properties 
+
+It is possible to store homogenous lists of simple values as properties. 
+For example, the following query creates a list from the `title` properties of the `Movie` nodes connected to `Keanu Reeves`. 
+It then sets that list as a `resume` property of `Keanu Reeves`.
+
 .Query
 [source, cypher]
 ----
-RETURN [x IN range(0,10) | x^3 ] AS result
+MATCH (keanu:Person {name: 'Keanu Reeves'})
+WITH keanu,[(keanu)-->(b:Movie) | b.title] AS movieTitles
+SET keanu.resume = movieTitles
+Return keanu.resume
 ----
 
 .Result
 [role="queryresult",options="header,footer",cols="1*<m"]
 |===
-| +result+
-| +[0.0,1.0,8.0,27.0,64.0,125.0,216.0,343.0,512.0,729.0,1000.0]+
+| +keanu.resume+
+| +["The Matrix Resurrections", "The Devils Advocate", "The Matrix", "The Replacements", "The Matrix Reloaded", "The Matrix Revolutions", "Johnny Mnemonic"]+
 1+d|Rows: 1
 |===
 
+It is not, however, possible to store heterogenous lists containing values of different data types as properties.
```

Comment:
a heterogenous list is one containing different values, so this sentence reads a little bit like you're repeating yourself, it is helpful to define what a heterogenous list is though! so perhaps rewording this could be useful :)

Reply:
We define it in section about property types, so I think it's fine to just say "It is not, however, possible to store heterogenous lists as properties."